Task1/ideal_weight.rb

```
@@ -0,0 +1,30 @@
+#ideal weight
+
+class Ideal
+  NUMBER = 110
+  def initialize
+    puts "Enter your name: "
+    @name = gets.chomp.capitalize
+    puts "Enter your height: "
+    @height = gets.chomp.to_f
+  end
+
+  def error_message
+    if @height.zero?
+      warn "incorrect input"
```

also, you can raise an exception here

raise AttributeError, 'Parameters must be positive numbers!'
https://rollbar.com/guides/ruby/how-to-raise-exceptions-in-ruby/

and handle it in the call method
```ruby
  def call
    validate_parameters
    area
  rescue AttributeError => e
    warn e
  end

---

yes, however in this moment i have fixed all problems and mistakes in my code, tested all possible inputs and It works much better now than ever.